sabnzbd/downloader.py

```diff
@@ -823,9 +810,13 @@ def run(self):
 
                     # Reset connection for new activity
                     nw.soft_reset()
-                    server.busy_threads.remove(nw)
-                    server.idle_threads.append(nw)
-                    self.remove_socket(nw)
+                    if server.article_queue and nw.connected:
+                        nw.article = server.article_queue.pop(0)
```

Comment:
Why pop(0)? It's less efdecient and doesn't really matter for us which article we get next? Or does it? 
https://stackoverflow.com/a/195839

Reply:
The difference is minimal for small lists and it helps the assembler a bit if the data arrive more or less in sequence. Here are the timings of doing pop(0), pop() and reading the elements using counting and index on a list with 20 items 1000000 times.
```
pop0
1.8746365000260994
pop
1.426083300029859
count
1.4542528000893071
```

Reply:
👍 👍